### PR TITLE
feat(telemetry): implement telemetry metrics

### PR DESCRIPTION
**Pull request by vbudhram**
Built from https://github.com/mozilla/fxa-discoverability-shield-study/pull/17 with updated telemetry metrics from https://github.com/mozilla/fxa-discoverability-shield-study/pull/10.

I split the commits into 3 to hopefully help with the review.

* First commit contains initial telemetry metrics
* Second commit contains control/treatment logic
* Third commit contains test plan, [rendered](https://github.com/mozilla/fxa-discoverability-shield-study/blob/3f47cfc83d3f5bafc4e8fb1e71e1b3c0a459985e/docs/TESTPLAN.md)

@philbooth I think this ready for review!

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -104,9 +104,15 @@ class StudyLifeCycleHandler {
    */
   async handleStudyEnding(ending) {
     console.log(`Study wants to end:`, ending);
-    for (const url of ending.urls) {
+    const user = await browser.fxa.getSignedInUser();
+    const studyInfo = await browser.study.getStudyInfo();
```
Comment (philbooth) [inline src/background.js]: Do you need the `await` here? The [docs](https://github.com/mozilla/shield-studies-addon-utils/blob/master/webExtensionApis/study/api.md#browserstudygetstudyinfo-) don't seem to mention `getStudyInfo` returning a promise.

File: src/feature.js [hunk 2]
```diff
@@ -102,18 +79,20 @@ class FxABrowserFeature {
...
+    this._state = "unverified";
     this.standardAvatar(UNVERIFIED_AVATAR);
-    browser.browserAction.setPopup({ popup: "popup/unverified/unverified.html" });
+    browser.browserAction.setPopup({ popup: UNVERIFIED_PAGE });
   }
 
   verifiedUser(user) {
-    if (user.avatar) {
+    this._state = "verified";
+    if (user.avatar && !user.avatarDefault) {
```
Comment (philbooth) [inline src/feature.js]: What are the conditions under which `user.avatar` and `user.avatarDefault` are both set? I didn't see that in local testing yet, interested just to understand and reproduce.

File: src/popup/fxa.js [hunk 3]
```diff
@@ -42,17 +67,16 @@ function setupAccountMenu(user) {
     if (emailElement) {
       emailElement.innerText = user.email;
 
-      if (user.avatar) {
-        document.getElementById("avatar").style.backgroundImage = `url("${user.avatar}")`;
-      } else {
+      if (user.avatarDefault) {
         document.getElementById("avatar").style.backgroundImage = `url("/icons/avatar.svg")`;
+      } else {
+        document.getElementById("avatar").style.backgroundImage = `url("${user.avatar}")`;
```
Comment (philbooth) [inline src/popup/fxa.js]: When I've tested locally, I've encountered conditions where both `avatar` and `avatarDefault` are `undefined`. Wouldn't that lead to us setting an empty `backgroundImage` here?

File: src/popup/fxa.js [hunk 4]
```diff
@@ -61,3 +85,7 @@ function openSyncPreferences() {
   browser.fxa.openSyncPreferences();
   window.close();
 }
+
+async function sendTelemetry(interactionType, elapsedTime) {
```
Comment (philbooth) [inline src/popup/fxa.js]: How come this is `async`? It doesn't have any `await`...

Comment (vbudhram) [reply src/popup/fxa.js]: Whoops, will remove.

File: src/popup/fxa.js (hunk 3 above)
Comment (vbudhram) [reply src/popup/fxa.js]: Strange, I haven't encountered that yet. I'll add a guard for it.

**Comment by vbudhram** [comment]
Thanks! @philbooth

**Pull request merged**